File: two-pointer/455-Assign-Cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        if len(s) == 0:
            return 0
        s.sort()
        g.sort()
        
        g_pointer = len(g) - 1
        s_pointer = len(s) - 1
        ans = 0
        while s_pointer > -1 and g_pointer > -1:
            if g[g_pointer] <= s[s_pointer]:
                s_pointer -= 1
                ans += 1
            g_pointer -= 1
        return ans

File: reference/two-pointer/455-Assign-Cookies.go
func findContentChildren(g []int, s []int) int {
    if len(s) == 0 {
\t\treturn 0
\t}

\tsort.Ints(g)
\tsort.Ints(s)

\tgPointer := len(g) - 1
\tsPointer := len(s) - 1
\tans := 0

\tfor gPointer >= 0 && sPointer >= 0 {
\t\tif g[gPointer] <= s[sPointer] {
\t\t\tans++
\t\t\tsPointer--
\t\t}
\t\tgPointer--
\t}

\treturn ans
}